### PR TITLE
[NOMERGE] Objective-C Date Support

### PR DESCRIPTION
**Pull request by dvdsgl**
- [x] Make date inference opt-in
- [x] Support marshaling `NSDate`
- [ ] Add tests

cc @davidhodge

### CODE REVIEW

**Comment by davidhodge** [comment]
Hurray!

**Comment by dvdsgl** [comment]
@davidhodge would you mind testing this branch?

**Comment by dvdsgl** [comment]
@davidhodge here's a web preview: https://app2.quicktype.io/#l=objc

**Comment by davidhodge** [comment]
Yes! If my in-flight WiFi works. About to take off.

File: src/Inference.ts [hunk 1]
```diff
@@ -124,13 +128,25 @@ export class TypeInference {
                     unionBuilder.addArray(cjson.getArrayForValue(value));
                     break;
                 case Tag.Date:
-                    unionBuilder.addStringType("date");
+                    if (this._inferDates) {
+                        unionBuilder.addStringType("date");
+                    } else {
```
Comment (schani) [inline src/Inference.ts]: This is not the way to do it.  We already have the string type mapping, which specifies which string types map to which.  That’s how, for example, in C# `data` and `time` maps to `date-time`, and for all other languages all date/time types map to `string`.  All you have to do is use a string type mapping that maps everything to `string`.

Comment (dvdsgl) [reply src/Inference.ts]: Ah ok, so I just need to pass the date inference flag into the target language.

Comment (schani) [reply src/Inference.ts]: I’ll have to take a look at the code. I’d rather not bother every target language with it.

Comment (dvdsgl) [reply src/Inference.ts]: But then this has to be repeated for each target language? I want an easy way to turn off date inference for any language.

**Comment by dvdsgl** [comment]
@davidhodge good thing quicktype's website works offline 😎

File: src/Inference.ts (hunk 1 above)
Comment (schani) [reply src/Inference.ts]: That's what I mean - I want to it to work across all target languages without having to put code into each target language.

Comment (schani) [reply src/Inference.ts]: Here's where you can override the string type mapping without meddling with target languages:

https://github.com/quicktype/quicktype/blob/master/src/index.ts#L134

**Comment by dvdsgl** [comment]
Closing until I have time to wrap this up.

**Pull request closed without merging**